### ISSUE TITLE
Signature of `ert--print-backtrace` has changed.

### ISSUE THREAD

**Issue by aaronbieber** [rejeep/ert-runner.el#47]
It appears that the function signature of the built-in `ert--print-backtrace` changed somewhere between Emacs 25.3 and Emacs 26.1, adding a second required argument. Consequently, some call within the plumbing of ERT is failing when using `ert-runner` in Emacs 26.1.

Note that if all of your tests are passing, you will encounter no problems. However, if you have a failing test, instead of seeing some helpful test failure information, you instead see `(wrong-number-of-arguments (2 . 2) 1)`. With the `--debug` flag, you will see the native Emacs backtrace ending on the frame `ert--print-backtrace(((...`.

Here is what it looks like in Emacs 26.1:

```
% EMACS=/usr/local/Cellar/emacs/26.1_1/Emacs.app/Contents/MacOS/EmacS cask exec ert-runner --debug
.Test commit-in-unsaved-buffer-fails backtrace:


Debugger entered--Lisp error: (wrong-number-of-arguments (2 . 2) 1)
  ert--print-backtrace(((t signal (error ("Periodic Commit cannot commit a file with no filename!")) nil) (t error ("Periodic Commit cannot c

<hundreds of lines of backtrace snipped>
```

And here is what is expected, which is running the same test suite with one failing test in Emacs 25.3. Note that most of the debug information here is relevant to the test suite I'm running, which is not important for this particular bug (and this is even rather useless since it is the expected output from the program):

```
% EMACS=/usr/local/Cellar/emacs/25.3/Emacs.app/Contents/MacOS/EmacS cask exec ert-runner
.Test commit-in-unsaved-buffer-fails backtrace:

  (if (not (buffer-file-name)) (error "Periodic Commit cannot commit a
  pcmm--commit(t)
  pcmm-commit()
  apply(pcmm-commit nil)
  (setq value-16 (apply fn-14 args-15))
  (unwind-protect (setq value-16 (apply fn-14 args-15)) (setq form-des
  (if (unwind-protect (setq value-16 (apply fn-14 args-15)) (setq form
  (let (form-description-18) (if (unwind-protect (setq value-16 (apply
  (let ((value-16 (quote ert-form-evaluation-aborted-17))) (let (form-
  (let ((fn-14 (function pcmm-commit)) (args-15 (list))) (let ((value-
  (save-current-buffer (set-buffer (get-buffer-create "empty buffer"))
  (pcmmt-create-proj "project" (save-current-buffer (set-buffer (get-b
  (let* ((pcmmt-dir (pcmmt-create-proj "project" (save-current-buffer
  (unwind-protect (let* ((pcmmt-dir (pcmmt-create-proj "project" (save
  (lambda nil (unwind-protect (let* ((pcmmt-dir (pcmmt-create-proj "pr
  ert--run-test-internal([cl-struct-ert--test-execution-info [cl-struc
  ert-run-test([cl-struct-ert-test commit-in-unsaved-buffer-fails "You
  ert-run-or-rerun-test([cl-struct-ert--stats (and t) [[cl-struct-ert-
  ert-run-tests((and t) (lambda (event-type &rest event-args) (cond ((
  ert-runner/run-tests-batch((and t))
  (let ((stats (ert-runner/run-tests-batch selector))) (kill-emacs (if
  ert-runner/run-tests-batch-and-exit((and t))
  (if ert-runner-verbose (ert-runner/run-tests-batch-and-exit ert-runn
  (let ((test-files (ert-runner--test-files tests)) (test-helper (f-ex
  ert-runner/run()
  apply(ert-runner/run nil)
  commander--handle-command(nil)
  commander-parse(nil)
  (if commander-parsing-done nil (commander-parse (or commander-args (
  eval-buffer(#<buffer  *load*> nil "/Users/airborne/dotfiles/configs/
  load-with-code-conversion("/Users/airborne/dotfiles/configs/emacs.d/
  load("/Users/airborne/dotfiles/configs/emacs.d/periodic-commit-minor
  command-line-1(("-scriptload" "/Users/airborne/dotfiles/configs/emac
  command-line()
  normal-top-level()

Test commit-in-unsaved-buffer-fails condition:

    (error "Periodic Commit cannot commit a file with no filename!")

<snip>
```

One way to locally fix this is to force `ert-compat.el` to load, since it brings along its own `ert--print-backtrace` that expects a single argument, which overrides the built-in one.

As a new user of both ERT and `ert-runner`, I banged my head against my desk for hours trying to figure out why this one test was getting stuck on this backtrace call until I discovered that it is because the test was failing for a valid reason, but spitting out this useless internal backtrace instead.

Let me know if I can be helpful at all; I just happen to have these two versions of Emacs installed, so I haven't tracked down the precise change that brought this about, nor have I picked through the guts of `ert-runner` to try to find the source of the problem.

**Comment by aaronbieber** [comment]
Odd, I see that this was actually addressed and closed in #42. It appears that I have a different version of `ert-runner`, although both my file (`ert-runner.el`) and the one here in this master branch are labeled 0.7.0.

**Comment by aaronbieber** [comment]
It looks like this is happening because Cask, by default, selects the latest tag to install. The last tag added to this project was back in 2015. This can be fixed by adding the repo path to your Cask file (like `(depends-on "ert-runner" :git "https://github.com/rejeep/ert-runner.el")`).

Maybe it's time to cut a release? Anyway, I'll close this since it's a Cask thing.
